nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/QuerySplunkIndexingStatus.java

```diff
@@ -194,7 +197,11 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 final EventIndexStatusResponse splunkResponse = unmarshallResult(responseMessage.getContent(), EventIndexStatusResponse.class);
 
                 splunkResponse.getAcks().entrySet().forEach(result -> {
-                    final FlowFile toTransfer = undetermined.get(result.getKey());
+                    FlowFile toTransfer = undetermined.get(result.getKey());
+                    if (!extractBoolean(toTransfer.getAttribute(ACKCHECKED_ATTRIBUTE)).orElse(Boolean.FALSE)) {
+                        toTransfer = setAckCheckedToTrue(session, toTransfer);
```

Comment:
This extraction logic looks to be the same as above. It would be luckier to not copy it (for the case we want to change the behaviour). I would suggest some method which simply tells if a flow file has the flag set. The value is irrelevant now anyway. Something like: `private static boolean isAlreadyChecked(FlowFile ff)`.

Reply:
Thanks, extracted it as proposed